File: Formvalidation.py
from PostgresConnector import PostgresConnector
import traceback
import psycopg2
class Formvalidation:


	def validate_data(self,vehicle_type,toll_type,date,price,vehicle_no):
		if vehicle_type.strip() == '' or toll_type.strip() == '' or price.strip() == '' or date.strip() == '' or vehicle_no.strip() =='':
			# figure out what to return here!
			raise Exception('input data has nulls')
		else:
			try:
				conn = PostgresConnector().get_connection()
				cursor = conn.cursor()
				query = """ INSERT INTO transactions
					(vehicle_type,toll_type,timestamp,price,vehicle_no
						) values(%s,%s,\'%s\',%s,\'%s\')
					""" % (vehicle_type,toll_type,date,float(price),vehicle_no) 
				#print "Inserting data to table using the query %s" % (query,)
				cursor.execute(query)
				conn.commit()
				return 'Success'
			except psycopg2.IntegrityError as e:
				raise Exception(' Unique key constraint failed ')
			except Exception:	
				raise Exception(' Something else went wrong')

File: TicketGenie/src/FormValidation.java
import java.sql.Connection;
import java.sql.PreparedStatement;

public class FormValidation {

	public boolean validate_data(String vehicle_type, String toll_type,
			String date, String price, String vehicle_no) throws Exception {

		Boolean flag = false;
		Connection conn = null;

		if (vehicle_type.trim().isEmpty() || toll_type.trim().isEmpty()
				|| date.trim().isEmpty() || price.trim().isEmpty()
				|| vehicle_no.trim().isEmpty()) {

			throw new Exception("Input Data has nulls");

		} else {

			try {
				DBConnector dbObj = new DBConnector();
				conn = dbObj.getConnection();
				PreparedStatement insertStmt = conn
						.prepareStatement("INSERT INTO transactions(vehicle_type,toll_type,timestamp,price,vehicle_no values (?,?,?,?,?)");
				int parameterPlaceHolder = 1;
				insertStmt.setString(parameterPlaceHolder++, vehicle_type);
				insertStmt.setString(parameterPlaceHolder++, toll_type);
				insertStmt.setString(parameterPlaceHolder++, date);
				insertStmt.setString(parameterPlaceHolder++, price);
				insertStmt.setString(parameterPlaceHolder++, vehicle_no);
				insertStmt.executeUpdate();
				flag = true;
			} catch (Exception e) {
				System.out.println("Exception while inserting data");
				e.printStackTrace();
			} finally {
				conn.close();
			}
		}

		return flag;
	}
}
